compiler/tests/fixtures/address_bytes10_map.fe

```diff
@@ -2,7 +2,7 @@ contract Foo:
     pub bar: map<address, bytes[10]>
 
     pub def read_bar(key: address) -> bytes[10]:
-        return self.bar[key]
```

Comment:
I think it would be good to add cases like this to our `compile_errors` tests to ensure they don't accidentally compile to something with undefined behavior. 

Am I right to assume that an assignment without `to_mem()` would now also be a compile error?

```
pub def failed_assignment(key: address):
    won_work: bytes[10 = self.bar[key]
```

I assume the above would fail and this would be needed to fix it.

```
pub def working_assignment(key: address):
    should_work: bytes[10 = self.bar[key].to_mem()
```

If that's the case, then I think we should add these to the compile errors.

Reply:
Ok, I pulled down the source and to my surprise, both of these compile:

Without `to_mem()`
```
    pub def some(key: address):
        x: bytes[10]
        x = self.bar[key]
```

with `to_mem()`

```
    pub def some(key: address):
        x: bytes[10]
        x = self.bar[key].to_mem()
```

I would have assumed that the version without `to_mem()` should fail, no?

Reply:
> I would have assumed that the version without to_mem() should fail, no?

Yes, I need to add an error here:

https://github.com/ethereum/fe/blob/07c09e516e13bd49ad65da936930a9353688b728/compiler/src/yul/mappers/assignments.rs#L61-L65

right now `to_mem` is implicit.

Reply:
here actually: 

https://github.com/ethereum/fe/blob/07c09e516e13bd49ad65da936930a9353688b728/compiler/src/yul/mappers/assignments.rs#L40-L42